### ISSUE TITLE
GDScript 2.0: A variable definition cannot use the function defined below

### ISSUE THREAD

**Issue by dalexeev** [godotengine/godot#63438]
### Godot version

v4.0.alpha12.official [2c11e6d9e]

### System information

Kubuntu 22.04

### Issue description

When you initialize a member variable, you cannot call a method whose definition is after the variable's definition.

### Steps to reproduce

```gdscript
extends Node

# Error: Cannot infer the type of "a" variable because the initial value doesn't have a set type.
var a := f()

func f() -> int:
    return 42

# Ok.
var b := f()
```

**Comment by Calinou** [comment]
I can't reproduce this in 3.5.rc 51e2c1990, so this is a regression in 4.0.

```gdscript
extends Node

var a := f()

func f() -> int:
	return 42

var b := f()

func _ready():
	print(a)
	print(b)
```

Output

```
42
42
```

**Comment by dalexeev** [comment]
Duplicate of #58542.

**Comment by dalexeev** [comment]
Fixed by #69471.
